charts/opentelemetry-operator/templates/deployment.yaml

```diff
@@ -2,22 +2,22 @@ apiVersion: apps/v1
 kind: Deployment
 metadata:
   labels:
-    app.kubernetes.io/name: opentelemetry-operator
+    {{- include "opentelemetry-operator.labels" . | nindent 4 }}
     control-plane: controller-manager
   name: {{ template "opentelemetry-operator.name" . }}-controller-manager
   namespace: {{ .Release.Namespace }}
 spec:
   replicas: {{ .Values.replicaCount }}
   selector:
     matchLabels:
-      app.kubernetes.io/name: opentelemetry-operator
+      {{- include "opentelemetry-operator.selectorLabels" . | nindent 6 }}
```

Comment:
@edwintye Can you test a `helm upgrade` with these changes?  I want to make sure a user running `0.16.0` will be able to successfully run `helm upgrade` for this chart and not have the new labels added via this PR affect anything.

Reply:
Thanks for the suggestion, you are right in that `helm upgrade` would indeed run into issues due to the change in `selectorLabels`.  I have not considered that as we run `helm template` and `kubectl apply --force` internally.

The only way to make this a save change is to move the [L51 in helpers](https://github.com/edwintye/opentelemetry-helm-charts/blob/use-helper-labels/charts/opentelemetry-operator/templates/_helpers.tpl#L51)  `app.kubernetes.io/instance: {{ .Release.Name }}` from `selectorLabels` to `labels`.  This approach deviates a little bit from the other chart in this repo.  Alternatively, this is a "breaking change", or at a minimum require a change in their upgrade path.

Annoyingly the line `app.kubernetes.io/name: {{ include "opentelemetry-operator.name" . }}` is permitted in `helm upgrade` to my amusement as it is changes the name of all the component i.e. forces a recreate anyway.

Reply:
cc @dmitryax

Reply:
@dmitryax how do you want to handle this?

Reply:
I just made the change as per above so that the upgrade can be done while still adding the labels. More concretely, the `selectorLabel` is a single field such that we have (for example in the `service`)
```yaml
  selector:
    app.kubernetes.io/name: opentelemetry-operator
    control-plane: controller-manager
```

which allows the upgrade.  I am happy to preserve the upgrade path if that makes all our life easier :)

Reply:
Thanks, I'll take a look tomorrow.

Reply:
I believe moving `app.kubernetes.io/instance: {{ .Release.Name }}` out of selector labels will harm anyone who has installed the chart multiple times under different releases.  IDK if thats actually a risk for this chart though.

Reply:
The current selector labels don't use `app.kubernetes.io/instance` at all, e.g. in [service](https://github.com/open-telemetry/opentelemetry-helm-charts/blob/main/charts/opentelemetry-operator/templates/service.yaml#L21) and in [deployment](https://github.com/open-telemetry/opentelemetry-helm-charts/blob/main/charts/opentelemetry-operator/templates/deployment.yaml#L12). This can be verified in our generated examples where the selector labels has not changed.

I think what you are thinking of is `app.kubernetes.io/name: {{ include "opentelemetry-operator.name" . }}` where the person upgrading may decide to suddenly introduce `nameOverride`.  This does not cause the upgrade to fail because `metadata.name` uses the definition `"opentelemetry-operator.name"` so an upgrade with `nameOverride` will force all the resources to be re-created. 

BUT, I have found a problem while doing my little tests against `main` (that is not part of this PR and a bug in general).  If I install using
```shell
helm install otel open-telemetry/opentelemetry-operator --version 0.17.2 --set nameOverride=whatever --set admissionWebhooks.certManager.enabled=false
```

then the chart fails because volume mount for secret adjusts against `nameOverride` as per [deployment.yaml#L119](https://github.com/open-telemetry/opentelemetry-helm-charts/blob/main/charts/opentelemetry-operator/templates/deployment.yaml#L119) but the secret name does not as per [operator-webhook.yaml#L14](https://github.com/open-telemetry/opentelemetry-helm-charts/blob/main/charts/opentelemetry-operator/templates/admission-webhooks/operator-webhook.yaml#L14).

Reply:
@edwintye see https://github.com/open-telemetry/opentelemetry-helm-charts/tree/main/charts/opentelemetry-operator#tls-certificate-requirement and choose one of the ise cases.

Reply:
yes, and my claim is that if you use the third option `admissionWebhooks.certManager.enabled` with `nameOverride` at the same time the chart install fails because the name of the cert (secret to be exact) does not match up.

Reply:
Ah, my bad, you are right. I haven't really tried out that option. Would you like to contribute a fix for it in a different pr ?

Reply:
@edwintye sorry I lost track of this.  I agree that moving the `app.kubernetes.io/instance: {{ .Release.Name }}` out of `opentelemetry-operator.selectorLabels` will work since the helper isn't being used today.  The fact that the examples' selector labels don't change is proof of this I think.